src/storage/index.ts

```diff
@@ -25,21 +27,143 @@ export const deleteConferenceSettings = () => {
+        SETTINGS_KEY,
+        JSON.stringify([
+          ...parsed,
+          {
+            encoded,
+            active: false,
+            id: parsed.length + 1,
+          },
+        ])
+      );
+    }
+  } else {
+    localStorage.setItem(
+      SETTINGS_KEY,
+      JSON.stringify([{ id: 1, encoded, active: true }])
+    );
+  }
 };
 
-export const getSettings = (): AppSettings => {
+export const editSettings = (settings: AppSettings, id: number) => {
+  const encoded = Buffer.from(JSON.stringify(settings), "utf-8").toString(
+    "base64"
+  );
+
+  const str = localStorage.getItem(SETTINGS_KEY);
+  if (str) {
+    const parsed = JSON.parse(str);
+
+    // for edit:
+    const newData = parsed.map((el: saveSettingFormat) => {
+      if (el.id === id)
+        return {
+          id: el.id,
+          active: el.active,
+          encoded: encoded,
+        };
+      else return el;
+    });
+
+    localStorage.setItem(SETTINGS_KEY, JSON.stringify(newData));
+  }
+};
+export const setActiveSettings = (id: number) => {
   const str = localStorage.getItem(SETTINGS_KEY);
   if (str) {
-    const planText = Buffer.from(str, "base64").toString("utf-8");
-    return JSON.parse(planText) as AppSettings;
+    const parsed = JSON.parse(str);
+
+    // for edit:
+    const newData = parsed.map((el: saveSettingFormat) => {
+      if (el.id === id)
+        return {
+          id: el.id,
+          active: true,
+          encoded: el.encoded,
+        };
+      else
+        return {
+          id: el.id,
+          active: false,
+          encoded: el.encoded,
+        };
+    });
+
+    localStorage.setItem(SETTINGS_KEY, JSON.stringify(newData));
   }
-  return {} as AppSettings;
+};
+
+export const deleteSettings = (id: number) => {
+  const str = localStorage.getItem(SETTINGS_KEY);
+  if (str) {
+    const parsed = JSON.parse(str);
+
+    // for edit:
+    const newData = parsed.filter((el: saveSettingFormat) => el.id !== id);
+
+    localStorage.setItem(SETTINGS_KEY, JSON.stringify(newData));
+  }
+};
+
+export const getSettings = (): IAppSettings[] => {
+  const str = localStorage.getItem(SETTINGS_KEY);
+  if (str) {
+    const data: { active: boolean; encoded: string; id: number }[] =
+      JSON.parse(str);
+    const decoded: IAppSettings[] = data.map((el) => {
+      return {
+        active: el.active,
+        decoded: JSON.parse(
+          Buffer.from(el.encoded, "base64").toString("utf-8")
+        ),
+        id: el.id,
+      };
+    });
+    return decoded;
+  }
+  return [] as IAppSettings[];
+};
+
+export const getActiveSettings = (): IAppSettings => {
+  const str = localStorage.getItem(SETTINGS_KEY);
+  if (str) {
+    const data: { active: boolean; encoded: string; id: number }[] =
+      JSON.parse(str);
+    const decoded: IAppSettings[] = data.map((el) => {
+      return {
+        active: el.active,
+        decoded: JSON.parse(
+          Buffer.from(el.encoded, "base64").toString("utf-8")
```

Comment:
can we find the active one and decode for the active setting, let avoid decode all.

Reply:
✔